src/picongpu/include/particles/bremsstrahlung/ScaledSpectrum.hpp

```diff
@@ -0,0 +1,312 @@
+    }
+
+    /** differential cross section times energy loss
+     */
+    struct StoppingPowerIntegrand
+    {
+        const float_64 Ekin;
+        const ScaledSpectrum& scaledSpectrum;
+
+        StoppingPowerIntegrand(const float_64 Ekin, const ScaledSpectrum& scaledSpectrum) :
+            Ekin(Ekin), scaledSpectrum(scaledSpectrum) {}
+
+        template<typename T_State, typename T_W>
+        void operator()(const T_State &x, T_State &dxdW, T_W W) const
+        {
+            dxdW[0] = this->scaledSpectrum.dcs(this->Ekin, W / this->Ekin) * W;
+        }
+    };
+
+public:
+
+    /** Generate lookup tables
+     */
+    void init()
+    {
+        namespace pam = PMacc::algorithms::math;
+        namespace odeint = boost::numeric::odeint;
+
+        // there is a margin of one cell to make the linear interpolation valid for border cells.
+        this->dBufScaledSpectrum = MyBuf(
+            new PMacc::container::DeviceBuffer<float_X, DIM2>(
+                electron::NUM_SAMPLES_EKIN + 1,
+                electron::NUM_SAMPLES_KAPPA + 1));
+        this->dBufStoppingPower = MyBuf(
+            new PMacc::container::DeviceBuffer<float_X, DIM2>(
+                electron::NUM_SAMPLES_EKIN + 1,
+                electron::NUM_SAMPLES_KAPPA + 1));
+
+        PMacc::container::HostBuffer<float_X, DIM2> hBufScaledSpectrum(this->dBufScaledSpectrum->size());
+        PMacc::container::HostBuffer<float_X, DIM2> hBufStoppingPower(this->dBufStoppingPower->size());
+        hBufScaledSpectrum.assign(float_X(0.0));
+        hBufStoppingPower.assign(float_X(0.0));
+
+        PMACC_AUTO(curScaledSpectrum, hBufScaledSpectrum.origin());
+        PMACC_AUTO(curStoppingPower, hBufStoppingPower.origin());
+
+        const float_64 EMin_SI = electron::MIN_ENERGY_MeV * 1.0e3 * UNITCONV_keV_to_Joule;
+        const float_64 EMax_SI = electron::MAX_ENERGY_MeV * 1.0e3 * UNITCONV_keV_to_Joule;
+        const float_64 lnEMin = pam::log(EMin_SI / UNIT_ENERGY);
+        const float_64 lnEMax = pam::log(EMax_SI / UNIT_ENERGY);
+
+        typedef boost::array<float_64, 1> state_type;
+
+        for(uint32_t EkinIdx = 0; EkinIdx < electron::NUM_SAMPLES_EKIN; EkinIdx++)
+        {
+            for(uint32_t kappaIdx = 0; kappaIdx < electron::NUM_SAMPLES_KAPPA; kappaIdx++)
+            {
+                float_64 kappa = static_cast<float_64>(kappaIdx) /
+                                 static_cast<float_64>(electron::NUM_SAMPLES_KAPPA - 1);
+                if(kappa == 0.0)
+                    kappa = electron::MIN_KAPPA;
+
+                const float_64 lnE_norm = static_cast<float_64>(EkinIdx) /
+                                          static_cast<float_64>(electron::NUM_SAMPLES_EKIN - 1);
+                const float_64 Ekin = pam::exp(lnEMin + (lnEMax - lnEMin) * lnE_norm);
+
+                *curScaledSpectrum(EkinIdx, kappaIdx) = static_cast<float_X>(this->chi(Ekin, kappa));
```

Comment:
The index range is (in both dimensions) by one smaller then the buffer size. Thus the upper boarder of the buffer array is still zero. Is this intentional?